### DISCUSSION TITLE
stop `.bloop` directory from being created

### DISCUSSION THREAD

**Discussion by ignamartinoli** [scalameta/nvim-metals#460] [Q&A]
I'm using **Metals** with `nvim-lspconfig` as explained [here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#metals).

Every time I open a **Scala** file, a `.bloop` directory gets created in my `$HOME`.
Is there a way to stop this from happening or at least specify another place for it to be created?

**Comment by ckipp01** [comment]
Thanks for the question @ignamartinoli. Sometimes the default logic in nvim-lspconfig isn't enough to correctly fine the root. That's one of the extra things [nvim-metals](https://github.com/scalameta/nvim-metals) offers, which is the recommended way to use Metals with the built-in LSP support for Neovim.

**Reply by ignamartinoli** [reply]
I tried `nvim-metals` but it wasn't even receiving auto-suggestions for `println`.

This is what I configured:

```lua
use {
	'scalameta/nvim-metals',
	after = 'nvim-lspconfig',
	config = function ()
		vim.opt_global.completeopt = { 'menuone', 'noinsert', 'noselect' }
		vim.opt_global.shortmess:remove('F'):append('c')

		local meta = require 'metals'.bare_config()

		meta.settings = {
			showImplicitArguments = true,
			excludedPackages = { 'akka.actor.typed.javadsl', 'com.github.swagger.akka.javadsl' },
		}

		meta.capabilities = require('cmp_nvim_lsp').update_capabilities(vim.lsp.protocol.make_client_capabilities())

		vim.api.nvim_create_autocmd('FileType', {
			pattern = { 'sbt', 'scala' },
			callback = function ()
				require('metals').initialize_or_attach(meta)
			end
		})
	end,
	requires = { 'nvim-lua/plenary.nvim', 'hrsh7th/nvim-cmp' }
}
```

Do you happen to know why?

**Reply by ckipp01** [reply]
And then what did the autocmd look like to start metals?

**Reply by ignamartinoli** [reply]
Isn't this auto-command enough to start **Metals**?

```lua
vim.api.nvim_create_autocmd('FileType', {
	pattern = { 'sbt', 'scala' },
	callback = function ()
		require('metals').initialize_or_attach(meta)
	end
})
```

I copied it from the [example configuration](https://github.com/scalameta/nvim-metals/discussions/39)

**Reply by ckipp01** [reply]
Ah sorry I totally missed the autocmd in there. Yes, that should be enough.  I'm not positive what's going on. I' on holiday at the moment, so when I get back I'll try to take a closer look. I'm going to transfer this over to nvim-metals as well.

**Reply by ignamartinoli** [reply]
Hey, no problem, I'm currently on exams so there's no rush. Enjoy your vacations! :smile:
